crates/rpc/src/handlers/validator.rs

```diff
@@ -305,3 +337,39 @@ pub async fn post_validator_identities_from_state(
 
     Ok(HttpResponse::Ok().json(BeaconResponse::new(validator_identities)))
 }
+
+#[get("/beacon/states/{state_id}/validator_balances")]
+pub async fn get_validator_balances_from_state(
+    state_id: Path<ID>,
+    query: Query<IdQuery>,
+    db: Data<ReamDB>,
+) -> Result<impl Responder, ApiError> {
+    let state = get_state_from_id(state_id.into_inner(), &db).await?;
+    let pairs: Vec<(Validator, u64)> = state
+        .validators
+        .into_iter()
+        .zip(state.balances.into_iter())
+        .collect();
+
+    let validator_balances = build_validator_balances(&pairs, query.id.as_ref());
+
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(validator_balances)))
+}
+
+#[post("/beacon/states/{state_id}/validator_balances")]
+pub async fn post_validator_balances_from_state(
+    state_id: Path<ID>,
+    body: Json<IdQuery>,
+    db: Data<ReamDB>,
+) -> Result<impl Responder, ApiError> {
+    let state = get_state_from_id(state_id.into_inner(), &db).await?;
+    let pairs: Vec<(Validator, u64)> = state
+        .validators
+        .into_iter()
+        .zip(state.balances.into_iter())
+        .collect();
+
+    let validator_balances = build_validator_balances(&pairs, body.id.as_ref());
+
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(validator_balances)))
+}
```

Comment:
```suggestion
#[get("/beacon/states/{state_id}/validator_balances")]
pub async fn get_validator_balances_from_state(
    state_id: Path<ID>,
    query: Query<IdQuery>,
    db: Data<ReamDB>,
) -> Result<impl Responder, ApiError> {
    let state = get_state_from_id(state_id.into_inner(), &db).await?;
    Ok(HttpResponse::Ok().json(BeaconResponse::new(build_validator_balances(&state
        .validators
        .into_iter()
        .zip(state.balances.into_iter())
        .collect(), query.id.as_ref())))
}

#[post("/beacon/states/{state_id}/validator_balances")]
pub async fn post_validator_balances_from_state(
    state_id: Path<ID>,
    body: Json<IdQuery>,
    db: Data<ReamDB>,
) -> Result<impl Responder, ApiError> {
    let state = get_state_from_id(state_id.into_inner(), &db).await?;
    Ok(HttpResponse::Ok().json(BeaconResponse::new(build_validator_balances(&state
        .validators
        .into_iter()
        .zip(state.balances.into_iter())
        .collect(), body.id.as_ref()))))
}
```

Please just inline all of this